components/playlist/browser/resources/components/playerSeeker.tsx

```diff
@@ -0,0 +1,253 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+import styled from 'styled-components'
+import { color, font } from '@brave/leo/tokens/css'
+
+interface Props {
+  videoElementRef: React.RefObject<HTMLVideoElement>
+}
+
+const SeekerContainer = styled.div`
+  display: flex;
+  flex-direction: column;
+`
+
+const TimeContainer = styled.div`
+  display: grid;
+  grid-template-columns: auto auto;
+  justify-content: space-between;
+
+  color: ${color.text.secondary};
+  font: ${font.primary.xSmall.regular};
+`
+
+const StyledProgress = styled.progress`
+  --progress-background: #423eee; /* TODO(sko) Neither color.button.background or color.semantic.button.background exist for now */
+  --progress-bar-height: 12px;
+  --progress-stroke-thickness: 2px;
+
+  width: 100%;
+  height: var(--progress-bar-height);
+  appearance: none;
+  position: relative;
+
+  &::-webkit-progress-bar {
+    background: rgba(0, 0, 0, 0.1);
+    height: var(--progress-stroke-thickness);
+    border-radius: calc(var(--progress-stroke-thickness) * 0.5);
+    opacity: 1;
+    transform: translate(0, calc((var(--progress-bar-height) - 100%) / 2));
+  }
+  &::-webkit-progress-value {
+    border-radius: calc(var(--progress-stroke-thickness) * 0.5);
+    background: var(--progress-background);
+  }
+  // TODO(sko) Can we split this so that we can avoid creating new class too
+  // many times?
+  &::after {
+    content: '';
+    position: absolute;
+    top: 0;
+    left: calc(
+      ${p => (p.value && p.max ? +p.value / +p.max : 0) * 100}% +
+        var(--progress-bar-height) * -0.5
+    );
+    width: var(--progress-bar-height);
+    height: var(--progress-bar-height);
+    background: var(--progress-background);
+    border-radius: calc(var(--progress-bar-height) * 2);
+  }
+`
+
+function formatTime (time: number) {
+  const stringFormat = (t: number) => String(t).padStart(2, '0')
+  const hours = stringFormat(Math.floor(time / 3600))
+  const minutes = stringFormat(Math.floor((time % 3600) / 60))
+  const seconds = stringFormat(Math.floor(time % 60))
+  return `${hours}:${minutes}:${seconds}`
```

Comment:
Optional: We could hide the hours/minutes if there aren't any (and design is fine with it)
```suggestion
    const parts = []
    if (hours) parts.push(hours)
    if (minutes || parts.length) parts.push(minute)
    
    // We always want to display seconds
    parts.push(seconds);

  return parts.join(':')
```